backend/tests/test_api/test_robots_api.py

```diff
@@ -15,6 +19,84 @@ def test_get_robots(test_app: FastAPI):
         assert len(response.json()) == 2
 
 
+def test_create_robot_success(test_app: FastAPI, session: Session):
+    body: dict = {
+        "id": 0,
+        "name": "test_robot",
+        "model": "test_robot_model",
+        "serial_number": "test_robot_serial_number",
+        "host": "localhost",
+        "port": 3000,
+        "enabled": True,
+        "status": "available",
+        "capabilities": ["Image", "ThermalImage"],
+    }
+    with TestClient(test_app) as client:
+        response: Response = client.post(url="/robots", data=json.dumps(body))
+        assert response.status_code == HTTPStatus.OK.value
+        assert read_robot_by_id(db=session, robot_id=response.json()["robot_id"])
+
+
+@pytest.mark.parametrize(
+    "expected_status_code, body",
+    [
+        (
+            HTTPStatus.BAD_REQUEST.value,
+            {  # Capabilities contain an invalid inspection type which should cause BAD REQUEST
+                "id": 0,
+                "name": "test_robot",
+                "model": "test_robot_model",
+                "serial_number": "test_robot_serial_number",
+                "host": "localhost",
+                "port": 3000,
+                "enabled": True,
+                "status": "available",
+                "capabilities": [
+                    "Image",
+                    "ThermalImage",
+                    "this_is_an_invalid_inspection",
+                ],
+            },
+        ),
+        (
+            HTTPStatus.CONFLICT.value,
+            {  # Name is a robot that already exists and this should cause CONFLICT
+                "id": 0,
+                "name": "Harald",
+                "model": "test_robot_model",
+                "serial_number": "test_robot_serial_number",
+                "host": "localhost",
+                "port": 3000,
+                "enabled": True,
+                "status": "available",
+                "capabilities": ["Image", "ThermalImage"],
+            },
+        ),
+        (
+            HTTPStatus.UNPROCESSABLE_ENTITY.value,
+            {  # Port is not an integer which should cause a validation error, UNPROCESSABLE_ENTITY
+                "id": 0,
+                "name": "test_robot",
+                "model": "test_robot_model",
+                "serial_number": "test_robot_serial_number",
+                "host": "localhost",
+                "port": "This is not an integer and should cause validation error",
+                "enabled": True,
+                "status": "available",
+                "capabilities": ["Image", "ThermalImage"],
+            },
+        ),
+    ],
+)
+def test_create_robot_throws_exception(
```

Comment:
Thin the name can be a bit misleading. I expect that the test throws an exception and that we use pytest.raises when the name of the test is ...throws_exception. Perhaps calling it something with invalid input is more representative

Reply:
Renamed to `test_create_robot_fails_with_invalid_input`